tests/components/wallbox/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Test the Wallbox config flow."""
+from unittest.mock import patch
+
+import requests_mock
+from voluptuous.schema_builder import raises
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.wallbox import config_flow
+from homeassistant.components.wallbox.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.wallbox.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+
+async def test_show_set_form(hass: HomeAssistant) -> None:
+    """Test that the setup form is served."""
+    flow = config_flow.ConfigFlow()
+    flow.hass = hass
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.wallbox.config_flow.PlaceholderHub.authenticate",
```

Comment:
I suggest to mock the library function so PlaceHolderHub.authenticate() gets tested.